src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +68,95 @@ public void TestSignaturesV2_4_X()
+        {            
+            // Calling CreateDataFrame with schema
+            {
+                var data = new List<GenericRow>();
+                data.Add(new GenericRow(new object[] { "Alice", 20 }));
+                data.Add(new GenericRow(new object[] { "Bob", 30 }));
+
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("Name", new StringType()),
+                    new StructField("Age", new IntegerType())
+                });
+                DataFrame df = _spark.CreateDataFrame(data, schema);
+                ValidateDataFrame(data.Select(a => a.Values), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<string> _) without schema
+            {
+                var data = new List<string>(new string[] { "Alice", "Bob" });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new StringType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);                
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<int> _) without schema
+            {
+                var data = new List<int>(new int[] { 1, 2 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new IntegerType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<double> _) without schema
+            {
+                var data = new List<double>(new double[] { 1.2, 2.3 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new DoubleType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<bool> _) without schema
+            {
+                var data = new List<bool>(new bool[] { true, false });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new BooleanType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+        }
+
+        /// <summary>
+        /// Validates that the dataframe returned by the CreateDataFrame API is correct as per the
+        /// given data.
+        /// </summary>
+        /// <param name="data">Given data for the dataframe</param>
```

Comment:
this comment doesn't make sense

Reply:
How about ->
Validates the correctness of the dataframe given the input data.

Reply:
Can we remove the comments and change the signature as follow?
```
private void ValidateDataFrame(DataFrame actual, IEnumerable<object[]> expectedRows, StructType expectedSchema)
```